assets/src/components/mapPage/routePropertiesCard.tsx

```diff
@@ -0,0 +1,168 @@
+import React from "react"
+import { useRoute } from "../../contexts/routesContext"
+import {
+  RoutePattern,
+  RoutePatternId,
+  DirectionName,
+  DirectionId,
+  RouteId,
+  Route,
+  ByRoutePatternId,
+} from "../../schedule"
+import { Card } from "../card"
+import { RoutePill } from "../routePill"
+
+const sortRoutePatterns = (routePatterns: RoutePattern[]): RoutePattern[] =>
+  routePatterns.sort((rp1, rp2) => rp1.sortOrder - rp2.sortOrder)
+
+const DirectionPicker = ({
+  selectedRoutePattern,
+  routePatterns,
+  selectRoutePattern,
+  directionNames,
+}: {
+  selectedRoutePattern: RoutePattern
+  routePatterns: ByRoutePatternId<RoutePattern>
+  selectRoutePattern: (routePattern: RoutePattern) => void
+  directionNames: {
+    0: DirectionName
+    1: DirectionName
+  }
+}) => {
+  return (
+    <div className="m-route-properties-card__direction-picker">
+      {
+        // TODO: Should we always show both buttons? Some routes only go one direction?
+      }
+      {[0, 1].map((directionId) => (
+        <div key={directionId}>
+          <input
+            type="radio"
+            id={`direction-radio-${directionId}`}
+            name="direction"
+            defaultChecked={selectedRoutePattern.directionId === directionId}
+            onChange={() => {
+              const targetDirectionId = directionId === 0 ? 1 : 0
+              const targetRoutePattern =
+                sortRoutePatterns(Object.values(routePatterns)).find(
+                  (routePattern) =>
+                    routePattern.directionId === targetDirectionId
+                ) || null
+
+              if (targetRoutePattern) {
+                selectRoutePattern(targetRoutePattern)
+              }
+            }}
+          />
+          <label htmlFor={`direction-radio-${directionId}`}>
+            {directionNames[directionId as DirectionId]}
```

Comment:
**suggestion:** I like avoiding `as` wherever possible since that causes us to loose out on TypeScript's typechecking. I was able to rewrite things slightly as following and get it to pass:

```
  const directionIds: DirectionId[] = [0, 1]

  return (
    <div className="m-route-properties-card__direction-picker">
      {
        // TODO: Should we always show both buttons? Some routes only go one direction?
      }
      {directionIds.map((directionId: DirectionId) => (
        <div key={directionId}>
          <input
            type="radio"
            id={`direction-radio-${directionId}`}
            name="direction"
            defaultChecked={selectedRoutePattern.directionId === directionId}
            onChange={() => {
              const targetDirectionId = directionId === 0 ? 1 : 0
              const targetRoutePattern =
                sortRoutePatterns(Object.values(routePatterns)).find(
                  (routePattern) =>
                    routePattern.directionId === targetDirectionId
                ) || null

              if (targetRoutePattern) {
                selectRoutePattern(targetRoutePattern)
              }
            }}
          />
          <label htmlFor={`direction-radio-${directionId}`}>
            {directionNames[directionId]}
          </label>
        </div>
      ))}
    </div>
  )
```

It's maybe a bit more verbose but I wasn't able to figure out a syntactically-correct way of typing `[0, 1]` inline with the call to `.map`.